### DISCUSSION TITLE
When Kernel symbols are unavailable, user-space methods that call kernel-space methods are not reported

### DISCUSSION THREAD

**Discussion by AntonIvanov87** [async-profiler/async-profiler#1043] [Q&A] [answered]
Hi! Say there is a trace:
```
  ...
  [13] do_syscall_64_[k]
  [14] entry_SYSCALL_64_after_hwframe_[k]
  [15] send
  [16] io.netty.channel.unix.Socket.sendAddress
```
that consumes much CPU cycles.

If Kernel symbols are not available:
```
[WARN] Kernel symbols are unavailable due to restrictions. Try
  sysctl kernel.perf_event_paranoid=1
  sysctl kernel.kptr_restrict=0
```
such traces are reported at all, this creates an illusion that CPU is spend elsewhere.

Is there any workaround to show at least a truncated trace like this:
```
  [0] Kernel symbols are unavailable
  [1] send
  [2] io.netty.channel.unix.Socket.sendAddress
```
?

**Comment by apangin** [comment, marked as answer]
Thank you for pointing this out.
This is actually a defect. Please submit a [bug report](https://github.com/async-profiler/async-profiler/issues/new?assignees=&labels=bug&projects=&template=bug-report.yml&title=%28short+issue+description%29).

In the meantime, the workaround is to use `ctimer` event instead. See #855
